src/ostree.cc

```diff
@@ -23,6 +23,32 @@ OstreeSysroot *Ostree::LoadSysroot(const std::string &path) {
   return sysroot;
 }
 
+OstreeDeployment *Ostree::getStagedDeployment(const std::string &path, const std::string &ostree_os) {
+  OstreeSysroot *sysroot = Ostree::LoadSysroot(path);
+  GPtrArray *deployments = NULL;
+  OstreeDeployment *res = NULL;
+
+  const char *osname;
+  if (ostree_os.empty()) {
+    OstreeDeployment *merge_deployment = ostree_sysroot_get_merge_deployment(sysroot, NULL);
+    osname = ostree_deployment_get_osname(merge_deployment);
+  } else {
+    osname = ostree_os.c_str();
+  }
+
+  deployments = ostree_sysroot_get_deployments(sysroot);
+
+  for (unsigned int i = 0; i < deployments->len; i++) {
```

Comment:
I noticed that ostree admin status shows list of deployments where the topmost deployment is the deployment that will boot on reboot. Maybe we just can take the deployment with 0 index instead?